src/huntsman/pocs/scheduler/observation/base.py

```diff
@@ -214,6 +209,21 @@ def mark_exposure_complete(self):
         """ Explicitly mark the current exposure as complete. """
         self._current_exp_num += 1
 
+    def get_filter_name(self, camera_name):
+        """ Determine filter name based on camera if `filter_name` is dict,
+        otherwise just return `filter_name`.
+
+        Args:
+            camera_name (str): The name of the camera.
+        Returns:
+            str: The filter name.
+        """
+        # If filter names is a dict, use camera name as key
+        if isinstance(self.filter_name, abc.Mapping):
```

Comment:
Think we should log and raise a `PanError` here if there is a `KeyError` on the camera name